src/main/scala/io/iohk/ethereum/jsonrpc/EthMiningService.scala

```diff
@@ -78,26 +78,27 @@ class EthMiningService(
   def getWork(req: GetWorkRequest): ServiceResponse[GetWorkResponse] =
     consensus.ifEthash(ethash => {
       reportActive()
-      val bestBlock = blockchain.getBestBlock()
-      val response: ServiceResponse[GetWorkResponse] =
-        Task.parZip2(getOmmersFromPool(bestBlock.hash), getTransactionsFromPool).map { case (ommers, pendingTxs) =>
-          val blockGenerator = ethash.blockGenerator
-          val PendingBlockAndState(pb, _) = blockGenerator.generateBlock(
-            bestBlock,
-            pendingTxs.pendingTransactions.map(_.stx.tx),
-            consensusConfig.coinbase,
-            ommers.headers,
-            None
-          )
-          Right(
-            GetWorkResponse(
-              powHeaderHash = ByteString(kec256(BlockHeader.getEncodedWithoutNonce(pb.block.header))),
-              dagSeed = EthashUtils.seed(pb.block.header.number.toLong),
-              target = ByteString((BigInt(2).pow(256) / pb.block.header.difficulty).toByteArray)
+      blockchain.getBestBlock() match {
+        case Some(block) =>
+          Task.parZip2(getOmmersFromPool(block.hash), getTransactionsFromPool).map { case (ommers, pendingTxs) =>
+            val blockGenerator = ethash.blockGenerator
+            val PendingBlockAndState(pb, _) = blockGenerator.generateBlock(
+              block,
+              pendingTxs.pendingTransactions.map(_.stx.tx),
+              consensusConfig.coinbase,
+              ommers.headers,
+              None
             )
-          )
-        }
-      response
+            Right(
+              GetWorkResponse(
+                powHeaderHash = ByteString(kec256(BlockHeader.getEncodedWithoutNonce(pb.block.header))),
+                dagSeed = EthashUtils.seed(pb.block.header.number.toLong),
+                target = ByteString((BigInt(2).pow(256) / pb.block.header.difficulty).toByteArray)
+              )
+            )
+          }
+        case _ => Task.now(Left(JsonRpcError.BlockNotFound))
```

Comment:
responding "Block not found" to getWork request will be misleading as no one was requesting a block. Internal Server Error would be more appropriate IMO

Reply:
Please also make sure the error is logged somewhere.